xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -561,20 +570,64 @@ void CGUIDialogPVRTimerSettings::Save()
   const CDateTime now(CDateTime::GetCurrentDateTime());
   if (!bStartSet || !bEndSet)
   {
-    if (m_startLocalTime < now)
-      m_startLocalTime = now;
-
-    if (m_endLocalTime < m_startLocalTime)
+    if (m_timerType->SupportsStartEndTime() && //i.e. has start/end time field
+        m_timerType->IsRepeating())            //but has no start day / end day spinners
     {
-      CLog::Log(LOGWARNING, "CGUIDialogPVRTimerSettings::Save - end time is before start time! Setting to start time.");
-      m_endLocalTime = m_startLocalTime;
+      if (m_endLocalTime < m_startLocalTime)   //And the end clock is earlier than the start clock
+      {
+        CLog::Log(LOGDEBUG, "CGUIDialogPVRTimerSettings::Save - End before start! Adding a day.");
+        m_endLocalTime += CDateTimeSpan(1, 0, 0, 0);
+        if (m_endLocalTime < m_startLocalTime)
+        {
+          CLog::Log(LOGWARNING, "CGUIDialogPVRTimerSettings::Save - End before start. Setting end time to start time.");
+          m_endLocalTime = m_startLocalTime;
+        }
+      }
+      else if (m_endLocalTime > (m_startLocalTime + CDateTimeSpan(1, 0, 0, 0))) //or the duration is more than a day
+      {
+        CLog::Log(LOGDEBUG, "CGUIDialogPVRTimerSettings::Save - End > 1 day after start! Removing a day.");
+        m_endLocalTime -= CDateTimeSpan(1, 0, 0, 0);
+        if (m_endLocalTime > (m_startLocalTime + CDateTimeSpan(1, 0, 0, 0)))
+        {
+          CLog::Log(LOGWARNING, "CGUIDialogPVRTimerSettings::Save - End > 1 day after start. Setting end time to start time.");
+          m_endLocalTime = m_startLocalTime;
+        }
+      }
+    }
+    else if (m_endLocalTime < m_startLocalTime) //The end clock is earlier than the start clock
+    {
+      CLog::Log(LOGWARNING, "CGUIDialogPVRTimerSettings::Save - Specified recording end time < start time: expect errors!");
     }
 
     if (!bStartSet)
-      m_timerInfoTag->SetStartFromLocalTime(m_startLocalTime);
+    {
+      // don't change date if the > or < settings apply
+      if (IsDateInCoveredRange(m_startLocalTime))
+        m_timerInfoTag->SetStartFromLocalTime(m_startLocalTime);
+      else
+      {
+        CDateTime oldStartLocalTime = m_timerInfoTag->StartAsLocalTime();
+        SYSTEMTIME timerStartTime;
+        m_startLocalTime.GetAsSystemTime(timerStartTime);
+        SetTimeFromSystemTime(oldStartLocalTime, timerStartTime);
+        m_timerInfoTag->SetStartFromLocalTime(oldStartLocalTime);
+      }
```

Comment:
I do not understand the purpose of this else block. If you want to guarantee that m_timerInfoTag's start time is untouched in case value is out of covereded time range, why you just don't write to it? Isn't this the best way to ensure it is unchnaged? Short: Is the else block needed at all?